error.go

```diff
@@ -20,295 +20,233 @@
 package goa
 
 import (
-	"bytes"
-	"encoding/json"
 	"fmt"
-	"strconv"
 	"strings"
 )
 
-type (
-	// ErrorID is an enum listing the possible types of errors.
-	ErrorID int
-
-	// TypedError describes an error that can be returned in a HTTP response.
-	TypedError struct {
-		ID   ErrorID
-		Mesg string
-	}
-
-	// MultiError records multiple errors.
-	MultiError []error
-
-	// BadRequestError is the type of errors that result in a response with status code 400.
-	BadRequestError struct {
-		Actual error
-	}
+const (
+	errInvalidParamTypeID     = "goa.ipt"
+	errMissingParamID         = "goa.mpa"
+	errInvalidAttributeTypeID = "goa.iat"
+	errMissingAttributeID     = "goa.mat"
+	errInvalidEnumValueID     = "goa.iev"
+	errMissingHeaderID        = "goa.mhe"
+	errInvalidFormatID        = "goa.ifo"
+	errInvalidPatternID       = "goa.ipa"
+	errInvalidRangeID         = "goa.ira"
+	errInvalidLengthID        = "goa.ile"
+	errInvalidEncodingID      = "goa.ien"
+	errInternalID             = "goa.int"
```

Comment:
The idea is that "id" is used by programs while "title" helps humans. I agree that the constants are not necessary and could just be inlined but I do believe there is value in distinguishing between machine and human consumed error class descriptions (which is what `ID` and `Title` are).